packages/blade/src/components/Tabs/Tabs.native.tsx

```diff
@@ -0,0 +1,214 @@
+    }, {});
+};
+
+const getRoutes = (tabs: React.ReactElement[]) => {
+  return tabs.map((TabComponent, index) => ({
+    index,
+    title: TabComponent.props.children,
+    key: TabComponent.props.value,
+    value: TabComponent.props.value,
+    leading: TabComponent.props.leading,
+    trailing: TabComponent.props.trailing,
+  }));
+};
+
+type Route = {
+  value?: string;
+  index?: number;
+};
+
+const getRouteIndexFromValue = ({ value, routes }: { value?: string; routes: Route[] }) =>
+  routes.findIndex((route) => route.value === value);
+
+const getRouteValueFromIndex = ({ index, routes }: { index?: number; routes: Route[] }) =>
+  routes.find((route) => route.index === index)?.value!;
+
+const Tabs = ({
+  children,
+  defaultValue,
+  value,
+  onChange,
+  size = 'medium',
+  variant = 'bordered',
+  autoWidth = false,
+}: TabsProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const isFirstRender = useFirstRender();
+  const tabs = getTabs(children);
+  const panels = getTabPanels(children);
+  const routes = getRoutes(tabs);
+  const isFilled = variant === 'filled';
+
+  const [selectedValue, setSelectedValue] = useControllableState({
+    defaultValue,
+    value,
+    onChange: (value) => {
+      if (isFirstRender) return;
+      onChange?.(value);
+    },
+  });
+
+  const index = getRouteIndexFromValue({
+    value: selectedValue,
+    routes,
+  });
+
+  const setIndex = React.useCallback(
+    (index: number) => {
+      const value = getRouteValueFromIndex({ index, routes });
+      setSelectedValue(() => value);
+    },
+    [routes, setSelectedValue],
+  );
+
+  const isVertical = false;
+  const contextValue = {
+    baseId: '',
+    selectedValue,
+    isVertical,
+    size,
+    variant,
+    autoWidth,
+    setSelectedValue: setIndex,
+  };
+
+  // Set initial value
+  React.useLayoutEffect(() => {
+    if (selectedValue) return;
+    setIndex(0);
+  }, [selectedValue, setIndex]);
+
+  const renderTabLabel = React.useCallback(
+    ({ route, focused }) => {
+      const { title, leading, trailing } = route;
+      const selectedState = focused ? 'selected' : 'unselected';
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      const validatedTrailingComponent = useTabsItemPropRestriction(trailing, size);
+
+      return (
+        <StyledTabButton autoWidth={!autoWidth && !isFilled} variant={variant} size={size}>
+          <Box display="flex" alignItems="center" flexDirection="row" gap="spacing.3">
+            {leading
+              ? React.cloneElement(leading as React.ReactElement, {
+                  size: iconSizeMap[size],
+                  color: `surface.action.icon.default.lowContrast`,
+                })
+              : null}
+            <Text
+              color={textColor[selectedState].default}
+              size={size === 'medium' ? 'medium' : 'large'}
+              weight={size === 'medium' ? 'bold' : 'regular'}
```

Comment:
Save as mapping somewhere. Don't hardcode it here directly. I'm assuming same will be applicable for web too

Reply:
Don't actually need to weight wala mapping. It's always bold.